packages/aws-cdk-lib/aws-elasticloadbalancingv2/lib/nlb/network-target-group.ts

```diff
@@ -280,11 +280,17 @@ export class NetworkTargetGroup extends TargetGroupBase implements INetworkTarge
         `Must be one of [${NLB_PATH_HEALTH_CHECK_PROTOCOLS.join(', ')}]`,
       ].join(' '));
     }
-    if (healthCheck.timeout && healthCheck.timeout.toSeconds() !== NLB_HEALTH_CHECK_TIMEOUTS[healthCheck.protocol]) {
-      ret.push([
-        'Custom health check timeouts are not supported for Network Load Balancer health checks.',
-        `Expected ${NLB_HEALTH_CHECK_TIMEOUTS[healthCheck.protocol]} seconds for ${healthCheck.protocol}, got ${healthCheck.timeout.toSeconds()}`,
-      ].join(' '));
+
+    const lowHealthCheckTimeout = 2;
+    const highHealthCheckTimeout = 120;
+    if (healthCheck.timeout) {
```

Comment:
Could we update the timeout property of the HealthCheck interface to reflect this new range?

Reply:
I understand that you mentioned the interface below.

https://github.com/aws/aws-cdk/blob/327035644b94ccee354cf34e1665f6eedfbf939d/packages/aws-cdk-lib/aws-elasticloadbalancingv2/lib/shared/base-target-group.ts#L110-L114

Please let me know if my understanding is wrong.
Thanks

Reply:
Yep, you got it! I looked at the timeout description in that interface and it looks like it is outdated. This PR would be a good time to update this and align with  the CloudFormation documentation. I think we could do something like this for the description: "The amount of time during which no response from a target means a failed health check. Must be 2 to 120 seconds". The default flag would also need to be updated. I think for this we could do the following:

`@default 6 seconds if the protocol is HTTP, 5 seconds if protocol is 'GENEVE', 30 seconds if target type is 'lambda', 10 seconds for TCP, TLS, or HTTPS`.

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-elasticloadbalancingv2-targetgroup.html#cfn-elasticloadbalancingv2-targetgroup-healthchecktimeoutseconds

Reply:
Thanks a lot for your kind suggestion!
I will align with the CFn docs